src/test/recovery/t/121_streaming_and_archiving.pl

```diff
@@ -0,0 +1,222 @@
+});
+
+# After switching wal, the current wal file will be marked as ready to be archived on the primary. But this wal file
+# won't get archived because of the incorrect archive archive_command
+my $walfile_ready = "pg_wal/archive_status/$current_walfile.ready";
+my $walfile_done = "pg_wal/archive_status/$current_walfile.done";
+
+# Wait for the standby to catch up
+$node_primary->wait_for_catchup($node_standby, 'replay',
+	$node_primary->lsn('insert'));
+
+# Wait for archive failure
+$node_primary->poll_query_until('postgres',
+q{SELECT failed_count > 0 FROM pg_stat_archiver}, 't')
+or die "Timed out while waiting for archiving to fail";
+
+# Due to archival failure, check if primary has only .ready wal file and not .done
+# Check if standby has .done created as wal_sender_archiving_status_interval is disabled.
+ok( -f "$primary_data/$walfile_ready",
+        ".ready file exists on primary for WAL segment $current_walfile"
+);
+ok( !-f "$primary_data/$walfile_done",
+        ".done file does not exist on primary for WAL segment $current_walfile"
+);
+
+ok( !-f "$standby_data/$walfile_ready",
+		".ready file does not exist on standby for WAL segment $current_walfile when wal_sender_archiving_status_interval=0"
+);
+ok(-f "$standby_data/$walfile_done",
+        ".done file exists on standby for WAL segment $current_walfile when wal_sender_archiving_status_interval=0"
+);
+
+##################### Test with wal_sender_archiving_status_interval enabled #####################
+$node_primary->adjust_conf('postgresql.conf', 'wal_sender_archiving_status_interval', "50ms");
+$node_primary->reload;
+# We need to enable wal_sender_archiving_status_interval on the standby as well. Technically the value doesn't matter
+# since the wal_receiver process only cares about this guc being set and not the actual value. Hence setting it to 100s
+# to reiterate this behavior
+$node_standby->adjust_conf('postgresql.conf', 'wal_sender_archiving_status_interval', "50ms");
+$node_standby->reload;
+
+$node_primary->safe_psql(
+	'postgres', q{
+	CREATE TABLE test2 AS SELECT generate_series(1,10) AS x;
+});
+
+my $current_walfile2 = $node_primary->safe_psql('postgres', "SELECT pg_walfile_name(pg_current_wal_lsn())");
+my $walfile_ready2 = "pg_wal/archive_status/$current_walfile2.ready";
+my $walfile_done2 = "pg_wal/archive_status/$current_walfile2.done";
+
+$node_primary->safe_psql(
+	'postgres', q{
+	SELECT pg_switch_wal();
+	CHECKPOINT;
+});
+
+# Wait for standby to catch up
+$node_primary->wait_for_catchup($node_standby, 'replay',
+	$node_primary->lsn('insert'));
+
+# Check if primary and standby have created .ready file for the wal that failed to archive
+ok( -f "$primary_data/$walfile_ready2",
+		".ready file exists on primary for WAL segment $current_walfile2 when wal_sender_archiving_status_interval=50ms"
+);
+ok( -f "$standby_data/$walfile_ready2",
+		".ready file exists on standby for WAL segment $current_walfile2 when wal_sender_archiving_status_interval=50ms"
```

Comment:
Need to remove extra tab.

```suggestion
	".ready file exists on primary for WAL segment $current_walfile2 when wal_sender_archiving_status_interval=50ms"
);
ok( -f "$standby_data/$walfile_ready2",
	".ready file exists on standby for WAL segment $current_walfile2 when wal_sender_archiving_status_interval=50ms"
```